queue/priority_queue.go

```diff
@@ -0,0 +1,66 @@
+package queue
+
+import (
+	"container/heap"
+)
+
+type PriorityQueueItem interface {
+	Less(other PriorityQueueItem) bool
+}
+
+type priorityQueue []PriorityQueueItem
+
+// Len returns the length of the ExpiryQueue.
+func (pq priorityQueue) Len() int { return len(pq) }
```

Comment:
There were some comments on this commit that I cherry-picked in https://github.com/lightningnetwork/lnd/pull/3415. You may want to replace this commit by the updated version.

Reply:
Fixed the godocs from your commit.